services/transactions.service.js

```diff
@@ -0,0 +1,137 @@
+import Transaction from "../models/transactions.model.js";
+import Account from "../models/accounts.model.js";
+import Category from "../models/categories.model.js";
+
+const transactionsService = {
+  getTransactions: async (req, res) => {
+    try {
+      if (await transactionsService.isAccountValid(req, res)) {
+        const transactions = await Transaction.find({
+          accountId: req.accountId,
+        });
+        res.status(200).json(transactions);
+      } else {
+        res.status(400).json({
+          message: "Account, whose transactions are requested, does not exist.",
+        });
+      }
+    } catch (err) {
+      res.status(500).json(err);
+    }
+  },
+  insertTransaction: async (req, res) => {
+    try {
+      if (await transactionsService.isAccountValid(req, res)) {
+        const transaction = new Transaction();
+        transaction.accountId = req.accountId;
+        transaction.title = req.body.title;
+        transaction.type = req.body.type;
+        transaction.date = Date.parse(req.body.date);
+        transaction.amount = req.body.amount;
+        transaction.category = await transactionsService.getCategoryId(
+          req,
+          res
+        );
+
+        transaction.save((err, doc) => {
+          if (err) res.status(500).json(err);
+          else res.status(201).json({ message: "Transaction created." });
+        });
+      } else {
+        res.status(400).json({
+          message:
+            "Account for which a transaction is being created, does not exist.",
+        });
+      }
+    } catch (err) {
+      res.status(500).json(err);
+    }
+  },
+
+  updateTransaction: async (req, res) => {
```

Comment:
Why do you need to update transactions? In my understanding are atomic records. They   can be only created (not updated and deleted). If you need to cancel created transaction - just create new one with negative operator.
But if you dont agree with me do how you see it, I can be wrong too